### ISSUE TITLE
Confusion Matrix for W&B

### ISSUE THREAD

**Issue by AndreaCossu** [ContinualAI/avalanche#564]
We should try to see how much code changes is actually needed to implement the builtin confusion matrix for W&B Logger.

Some points to note:

1. Metrics are meant to be disconnected from the loggers, so it is not ideal to have two different implementation of the CM. It is also true that the current one provides the CM directly, while the other would provide information on logits, true labels etc which combined can be used to produce the CM. Alas, there is no way around this if we want to support the builtin CM.
2. We could create additional CM metrics for W&B and expose only the usual utility function to build the metrics with the param `wandb=True/False`. 
3. We have to assess the impact on the existing loggers

I'll make a draft PR to sketch an implementation and we will see how much this violates the Avalanche design.
